opensrp-app/src/main/java/org/smartregister/sync/helper/TaskServiceHelper.java

```diff
@@ -95,6 +96,20 @@ public TaskServiceHelper(TaskRepository taskRepository) {
     public List<Task> fetchTasksFromServer() {
         Set<String> planDefinitions = getPlanDefinitionIds();
         List<String> groups = getLocationIds();
+
+        List<Task> tasks = batchFetchTasksFromServer(planDefinitions,groups);
+        int batchFetchCount = tasks.size();
+
+        while( tasks != null &&  batchFetchCount >= TASK_PULL_LIMIT) {
```

Comment:
since the limit can always be changed from the server, retry sync if any record was synced